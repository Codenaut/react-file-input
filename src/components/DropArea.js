import React from 'react';
import PropTypes from 'prop-types';

import '../styles/DropArea.scss';

const Droparea = ({ dragging, onDragEnter, onDragLeave, onDrop, openFileDialog }) => {
  const className = dragging
    ? `brainhub-drop-area  brainhub-drop-area--dragging`
    : `brainhub-drop-area`;

  return (
    <div
      className={className}
      onDragEnter={onDragEnter}
      onDragLeave={onDragLeave}
      onDrop={onDrop}
    >
      {dragging
        ? (<div className="brainhub-drop-area__info">Drop here to select file.</div>)
        : (<div className="brainhub-drop-area__button" role="button" tabIndex={-1} onClick={openFileDialog}>+ Add File</div>)
      }
    </div>
  );
};

Droparea.defaultProps = {
  dragging: false,
};

Droparea.propTypes = {
  dragging: PropTypes.bool,
  onDragEnter: PropTypes.func.isRequired,
  onDragLeave: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
  openFileDialog: PropTypes.func.isRequired,
};

export default Droparea;
