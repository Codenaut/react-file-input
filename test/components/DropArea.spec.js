import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DropArea from '../../src/components/DropArea';

const defaultProps = {
  onDragEnter: () => null,
  onDragLeave: () => null,
  onDrop: () => null,
  openFileDialog: () => null,
};

const setup = (props = {}) => {
  const dropArea = shallow(<DropArea {...defaultProps} {...props} />);

  return {
    dropArea,
  };
};

describe('components', () => {
  describe('DropArea', () => {
    it('should render an add button by default', () => {
      const { dropArea } = setup();

      const button = dropArea.find('.brainhub-drop-area__button');
      const info = dropArea.find('.brainhub-drop-area__info');

      expect(button).toHaveLength(1);
      expect(info).toHaveLength(0);

      expect(button.text()).toBe('+ Add File');
      expect(button.prop('role')).toBe('button');
      expect(button.prop('onClick')).toBe(defaultProps.openFileDialog);
    });

    it('should render an info message if it is dragging', () => {
      const { dropArea } = setup({ dragging: true });

      const button = dropArea.find('.brainhub-drop-area__button');
      const info = dropArea.find('.brainhub-drop-area__info');

      expect(button).toHaveLength(0);
      expect(info).toHaveLength(1);

      expect(info.text()).toBe('Drop here to select file.');
    });

    it('should render with custom class if defined', () => {
      const className = 'custom_class';
      const { dropArea } = setup({ className });

      expect(dropArea.find('div').first().hasClass(className)).toBeTruthy();
      expect(dropArea.find('div').first().hasClass('brainhub-drop-area')).toBeTruthy();

      dropArea.setProps({ dragging: true });

      expect(dropArea.find('div').first().hasClass(className)).toBeTruthy();
      expect(dropArea.find('div').first().hasClass('brainhub-drop-area')).toBeTruthy();
      expect(dropArea.find('div').first().hasClass('brainhub-drop-area--dragging')).toBeTruthy();
    });

    it('should match exact snapshot', () => {
      const tree = renderer.create(
        <div>
          <DropArea {...defaultProps} />
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('should match exact snapshot when dragging', () => {
      const tree = renderer.create(
        <div>
          <DropArea {...defaultProps} dragging />
        </div>
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
