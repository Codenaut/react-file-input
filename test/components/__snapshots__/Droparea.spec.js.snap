// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components DropArea should match exact snapshot 1`] = `
<div>
  <div
    className="brainhub-drop-area"
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDrop={[Function]}
  >
    <div
      className="brainhub-drop-area__button"
      onClick={[Function]}
      role="button"
      tabIndex={-1}
    >
      + Add File
    </div>
  </div>
</div>
`;

exports[`components DropArea should match exact snapshot when dragging 1`] = `
<div>
  <div
    className="brainhub-drop-area  brainhub-drop-area--dragging"
    onDragEnter={[Function]}
    onDragLeave={[Function]}
    onDrop={[Function]}
  >
    <div
      className="brainhub-drop-area__info"
    >
      Drop here to select file.
    </div>
  </div>
</div>
`;
